### PR TITLE
[receiver/splunk_hec] Fix reusing the shared splunkhecreiver

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Reuse the shared splunkhec receiver between logs and metrics.

**Testing:**
Unit test.

### CODE REVIEW

File: receiver/splunkhecreceiver/factory_test.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ package splunkhecreceiver
 
 import (
 	"context"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent"
```
Comment (fatsheep9146) [inline receiver/splunkhecreceiver/factory_test.go:8]: This line maybe put at last is better?

**Review by fatsheep9146 — APPROVED** [review]

File: receiver/splunkhecreceiver/factory.go [hunk 2]
```diff
@@ -68,6 +68,7 @@ func createMetricsReceiver(
 	if err != nil {
 		return nil, err
 	}
+	r.Component.(*splunkReceiver).metricsConsumer = consumer
```
Comment (dmitryax) [inline receiver/splunkhecreceiver/factory.go:71]: Just for readability 
```suggestion
	r.Unwrap().(*splunkReceiver).metricsConsumer = consumer
```

File: receiver/splunkhecreceiver/factory.go [hunk 3]
```diff
@@ -88,6 +89,7 @@ func createLogsReceiver(
 	if err != nil {
 		return nil, err
 	}
+	r.Component.(*splunkReceiver).logsConsumer = consumer
```
Comment (dmitryax) [inline receiver/splunkhecreceiver/factory.go:92]:
```suggestion
	r.Unwrap().(*splunkReceiver).logsConsumer = consumer
```

File: receiver/splunkhecreceiver/factory.go (hunk 2 above)
Comment (dmitryax) [reply receiver/splunkhecreceiver/factory.go]: It cannot be unexported, it needs to inherit the `Component` interface implementation

**Review by dmitryax — APPROVED** [review]

**Pull request merged**